ftw/linkchecker/command/checking_links.py

```diff
@@ -1,35 +1,86 @@
+    plone_sites_information = [{
+        'id': plone_site.getId(),
+        'path': '/'.join(plone_site.getPhysicalPath()),
+        'title': plone_site.Title()
+    } for plone_site in plone_sites]
 
-    # Set up request for debug / bin/instance run mode.
-    app = makerequest(plone)
-    setRequest(app.REQUEST)
+    return plone_sites_information
 
-    houptding = app.restrictedTraverse('/')
 
-    # here all pages need to be found
-    # for now there is only 'Plone'
-    # later on we need to iterate through
-    # each page and send reports to different people
+def setup_plone(plone_site):
+    user = AccessControl.SecurityManagement.SpecialUsers.system
+    user = user.__of__(plone_site.acl_users)
+    newSecurityManager(plone_site, user)
+    setSite(plone_site)
 
-    MINI_SITE = houptding.get('Plone')
 
-    user = AccessControl.SecurityManagement.SpecialUsers.system
-    user = user.__of__(MINI_SITE.acl_users)
-    newSecurityManager(MINI_SITE, user)
+def get_relation_catalog_for_plone_site(app, PloneSiteId):
+    plone_site = app.get(PloneSiteId)
+    setup_plone(plone_site)
+
+    return getUtility(ICatalog)
+
+
+def catalog_lookup(relation_catalog):
+    x = relation_catalog
+    relations = list(x.findRelations())
+    print(relations)
+
+
+def get_external_broken_links_in_plone_page(app, site_info):
+    # this function needs to find all external links in the plone site
+    # then run the script linkchecker from the branch mo/external_link_fetcher
+    return {}
+
+
+def get_broken_relations_in_plone_page(app, site_info):
+    relation_catalog = get_relation_catalog_for_plone_site(
+        app,
+        site_info['id']
+    )
+    catalog_lookup(relation_catalog)
+
+
+def create_excel_report_and_return_filepath(
+        site_info, broken_internal, broken_external):
+    # this function needs to create an excel report, safe it into a temp folder
+    # it then returns the path to that file.
+    # It makes use of the report generator from branch mo/excel_report_generator
+    pass
+
+
+def send_mail_with_excel_report_attached(email_address, path_to_report):
+    # this function sends the previously generated excel report to the
+    # responsible email address with a simple info body.
+    # It makes use of the report mailer from branch mo/mail_sending
+    pass
 
-    setSite(MINI_SITE)
 
-    relation_catalog = getUtility(ICatalog)
+def create_and_send_mailreport_to_plone_site_responible_person(
+        email_address, site_info, broken_internal, broken_external):
+    path_to_report = create_excel_report_and_return_filepath(
+        site_info, broken_internal, broken_external)
+    send_mail_with_excel_report_attached(email_address, path_to_report)
 
 
-if __name__ == '__main__':
-    main()
+def main(app, *args):
+    plone_sites_information = get_plone_sites_information(app)
+    for site_info in plone_sites_information:
+        email_address = 'hugo.boss@4teamwork.ch'
```

Comment:
Is it planned to put this in some sort of configuration file?

Reply:
I opened Issue #15.
I also asked about that myself here: https://github.com/4teamwork/ftw.linkchecker/pull/14#discussion_r235780887